### DISCUSSION TITLE
How to disable click Next in wizard step?

### DISCUSSION THREAD

**Discussion by hiimnhan** [cloudscape-design/components#632] [Q&A] [answered]
I want to prevent user from clicking Next button when they haven't filled all thing in that step yet, I don't see any props or feature in the wizard for that. How should I do that?

**Comment by connorlanigan** [comment, marked as answer]
Disabling the Next button is not supported. Instead, we recommend to follow [our Validation pattern](https://cloudscape.design/patterns/general/validation/), where the primary button stays always enabled and is used to trigger validation. This helps the user understand what they need to do in order to proceed.

**Reply by hiimnhan** [reply]
Do I need to wrap any step in the wizard with the form element and use the Next button to validation that step?

**Reply by connorlanigan** [reply]
You don't need to wrap the Wizard with a form element/component. The Wizard component provides the `onNavigate` and `onSubmit` events that are called when the user clicks on the primary button of the Wizard, and you can use these events to perform your validation.

You can find a demo of the Wizard in a page in [the _"Multipage create"_ demo](https://cloudscape.design/examples/react/wizard.html). You can find [the source code for the demo](https://github.com/cloudscape-design/demos/tree/main/src/pages/wizard) via the link in the top right in the demo.
